NDTensors/ext/NDTensorsMetalExt/linearalgebra.jl

```diff
@@ -19,17 +19,24 @@ function NDTensors.Expose.ql_positive(A::Exposed{<:MtlMatrix})
   return adapt(unwrap_array_type(A), Matrix(Q)), adapt(unwrap_array_type(A), L)
 end
 
+using NDTensors.TypeParameterAccessors: set_type_parameters
 function LinearAlgebra.eigen(A::Exposed{<:MtlMatrix})
   Dcpu, Ucpu = eigen(expose(NDTensors.cpu(A)))
-  D = adapt(set_ndims(set_eltype(unwrap_array_type(A), eltype(Dcpu)), ndims(Dcpu)), Dcpu)
+  D = adapt(
+    set_type_parameters(unwrap_array_type(A), (eltype, ndims), (eltype(Dcpu), ndims(Dcpu))),
```

Comment:
This is implicitly `Base.ndims` because its in an extension. Should I be more explicit?

Reply:
I think it is fine as it is. Though we should keep that question in mind as we continue developing this code and be more explicit if it starts to get confusing and we can't keep track of which `ndims` we are using.